src/codec_aom.c

```diff
@@ -715,57 +682,95 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
                 cfg->monochrome = 1;
             }
         }
+    }
+
+    avifBool dimensionChanged = AVIF_FALSE;
+    if ((cfg->g_w != image->width) || (cfg->g_h != image->height)) {
+        cfg->g_w = image->width;
+        cfg->g_h = image->height;
```

Comment:
Seems just changing `g_w` and `g_h` here only works when sending smaller frames, due to aom writing first frame's dimension to stream header.

To support larger frames, seems we need user to provide the expect dimension before starting encoding, and set `g_forced_max_frame_width` and `g_forced_max_frame_height`. I'm going to implement this in another PR.

Reply:
That's correct. An AV1 bitstream begins with a sequence header OBU, which advertises the maximum frame width and maximum frame height. Then each frame has a frame header OBU that may optionally specify a frame width and height smaller than the maximum. So if we are encoding multiple frames with increasing frame widths and heights, we will need a way to specify the maximum frame width and maximum frame height before encoding the first frame.